addons/sale/models/sale.py

```diff
@@ -751,10 +751,13 @@ def _compute_invoice_amount(self):
             # Refund invoices linked to invoice_lines
             refund_invoices = invoice_lines.mapped('invoice_id.refund_invoice_ids').filtered(lambda inv: inv.state in ('open', 'paid'))
             # Total invoiced amount
-            invoiced_amount_total = sum(invoice_lines.mapped('price_total'))
+            from_currency = self[0].currency_id
```

Comment:
I would simply use `line.currency_id` in the list comprehension. You take advantage of the pre-fetching anyway.